osquery/main/windows/main.cpp

```diff
@@ -75,6 +59,45 @@ void DebugPrintf(const char* fmt, ...) {
   va_end(vl);
 }
 
+// A helper function to return a HANDLE to the named Stop Event for child
+// processes
+HANDLE getStopEvent() {
+  auto stopEvent =
+      ::OpenEventA(SYNCHRONIZE, FALSE, osquery::kStopEventName.c_str());
```

Comment:
Found the problem.  This needs to be:
```
  auto stopEvent =
      ::OpenEventA(SYNCHRONIZE | EVENT_MODIFY_STATE, FALSE, osquery::kStopEventName.c_str());
```
Basically, the call to `SetEvent` below is failing because we require `EVENT_MODIFY_STATE` perms to modify the state, so the main thread of the worker was never shutting down because it was never receiving the signal.

This changed from the original behavior, as in the original behavior the Event was being `Create`d and `Set` in the same process, thus the security attributes of the `HANDLE` to the event were higher level permissions so the `Set` would be successful. Woot! :D